script.js

```diff
@@ -148,10 +200,53 @@ function currencyDisplayTable(event) {
         //creating cells
         row.insertCell(0).textContent = currency.base;
         row.insertCell(1).textContent = quoteCurrency;
-        row.insertCell(2).textContent = currency.rates[quoteCurrency];
+        if (
+          specialRates[baseCurrency] &&
+          specialRates[baseCurrency][quoteCurrency]
+        ) {
+          const specialRate = specialRates[baseCurrency][quoteCurrency];
+          row.insertCell(2).textContent = `${currency.rates[quoteCurrency]} ${
+            currentRate >= specialRate
+              ? "Oh  :fire:  Don't trade today.The conversion rate is too high"
+              : ""
+          }`;
+        }
       }
     }
   });
   // Append the table to the grid container
   gridContainer.appendChild(table);
 }
+
+// checking for special rates and making alert
+function checkSpecialRates() {
```

Comment:
this works well, but it feels a bit overkill to call it every minute, also it can be disturbing for the user if there are multiple special offers (multiple alerts), I think it is easier to show an alert when the user adds a new rate in `brandCurrencyBeta`. Or you can delete this logic and keep only the one in `currencyDisplayTable` I think it's sufficient :) What do you think?

Reply:
Then I will delete that :).